.github/workflows/e2e-tests.yml

```diff
@@ -0,0 +1,12 @@
+# This workflow runs e2e tests as a Github Action for every pull request created against this repo
+name: e2e-tests
+on:
+  pull_request:
+  schedule:
```

Comment:
I don't see a benefit to running these nightly. Barring flakes, the results should be the same. It might be good to trigger this on pushes to master as well as on PR.